src/solver/optimisation/constraints/RampingIncreaseRate.cpp

```diff
@@ -0,0 +1,47 @@
+#include "RampingIncreaseRate.h"
+
+void RampingIncreaseRate::add(int pays, int cluster, int clusterIndex, int pdt, bool Simulation)
```

Comment:
You may define a class both for `RampingIncreaseRate` and `RampingDecreaseRate`. With a pure virtual method for behaviors that differ, e.g
```diff
-.NumberOfDispatchableUnits(cluster, -maxUpwardPowerRampingRate)
-.NumberStartingDispatchableUnits(cluster, -pminDUnGroupeDuPalierThermique)
+.NumberStoppingDispatchableUnits(cluster, pmaxDUnGroupeDuPalierThermique)
+.NumberBreakingDownDispatchableUnits(cluster, pmaxDUnGroupeDuPalierThermique)
```

See https://refactoring.guru/design-patterns/template-method

Reply:
Pending until the constraint equations are validated